src/app/modules/timesheet/components/timesheet-edit/timesheet-edit.component.ts

```diff
@@ -67,6 +70,15 @@ export class TimesheetEditComponent implements OnInit {
 
   onUserInput() {
     this.showLinks = false;
+    this.submitMessage = null;
+  }
+
+  setShortUrl(action: string) {
+    const getToken = (action === 'edit') ? this.timesheetService.getEditToken() : this.timesheetService.getReviewToken();
```

Comment:
A method `getToken` with a param would have been wiser, but it does not concern your PR, so forget ^^